float32 vx_r_val
float32 vy_r_val
